docs: More in depth explanations

ikks opened the pull request.

ikks committed:
docs: Starting point for minimode and helix mode

ikks commented:
There are other docs linked that are being written.  I expect to continue documenting while learning more about the internals to be as precise as possible and general to be synced with the project as a whole.

ikks committed:
Add footer for edition

ikks committed:
Add initial docs for command, editor, keybind, palette

ikks committed:
Add some more information to palettes

ikks committed:
Added more documentation to editor

ikks committed:
Added more documentation to palette

ikks committed:
Add explanation of exchanging data

ikks committed:
Links between developer docs

ikks commented:
TODO: in the near future is having some navigation in the contents of each page and also a general index allowing to navigate. 

Maybe a search thingy...

And some more CSS magic.

And of course,  a lot more content wherever feel is appropriate.

neurocyte commented:
You seem to be working on this still. Let me know when you want to merge it. Or perhaps mark it as draft.

ikks committed:
Marked editor as draft

All the docs contained in this group of commits are marked as not draft,
except for editor, which I will be working as I get more experience and
get more precise on the various topics related to the editor and groups
of functions.

ikks commented:
Hey @neurocyte thank you.  For now I guess the docs are good in content, I would like to continue improving the navigability and give some more love to the site. I'm not webdev myself, so not much of CSS or the like.

I can open issues for the items I mentioned earlier, to maybe put some help wanted on some of them.

I remember that in the Ghostty Discord they had a channel for screenshots, maybe, it would be a good idea to have a #channel for screencasts of less than 30 seconds to show off things that people is doing with the editor? if not a new channel, where would be a good place to put them and invite people to put them?   I would like to put in contributing, the screencasts way of showing others how to do things.

I have learnt a lot looking at the ones you have shared in the channels as well as in the Devlog.

ikks committed:
Add more internal links and precisions on command arguments

neurocyte commented:
> For now I guess the docs are good in content, I would like to continue improving the navigability and give some more love to the site.

These docs are really looking. Thanks for working on them.

neurocyte commented:
> I remember that in the Ghostty Discord they had a channel for screenshots, maybe, it would be a good idea to have a #channel for screencasts of less than 30 seconds to show off things that people is doing with the editor? if not a new channel, where would be a good place to put them and invite people to put them? I would like to put in contributing, the screencasts way of showing others how to do things.

This is a great idea. I think we should chat about it in discord get it setup.
